dff/pipeline/pipeline/actor.py

```diff
@@ -56,6 +57,79 @@ def error_handler(error_msgs: list, msg: str, exception: Optional[Exception] = N
         logger.error(msg, exc_info=exception)
 
 
+def types_match(type1: Type, type2: Type) -> bool:
+    """
+    This function compares types with assumption that one of the types might be a :py:class:`typing.ForwardRef`.
+    If it is so, it compares type and forward reference by name.
+
+    :param type1: First type to compare.
+    :param type2: Second type to compare.
+    :return: True if types are equal, False otherwise.
+    """
+    if type1 == type2:
+        return True
+    elif type(type1) is ForwardRef or type(type2) is ForwardRef:
+        type1_name = type1.__forward_arg__ if type(type1) is ForwardRef else type1.__name__
+        type2_name = type2.__forward_arg__ if type(type2) is ForwardRef else type2.__name__
+        return type1_name == type2_name
+    else:
+        return False
+
+
+def validate_callable(
+    callable: Callable,
+    name: str,
+    flow_label: str,
+    node_label: str,
+    error_msgs: list,
+    logging_flag: bool = True,
+    expected_types: Optional[Tuple[Type, ...]] = None,
+    return_type: Optional[Tuple[Type]] = None,
+):
+    """
+    This function validates a function during :py:class:`~dff.script.Script` validation.
+    It checks parameter number (unconditionally), parameter types (if specified) and return type (if specified).
+
+    :param callable: Function to validate.
+    :param name: Type of the function (label, condition, response, etc.).
+    :param flow_label: Flow label this function is related to (used for error localization only).
+    :param node_label: Node label this function is related to (used for error localization only).
+    :param error_msgs: List that contains error messages. All the error message will be added to that list.
+    :param logging_flag: The flag which defines whether logging is necessary. Defaults to `True`.
+    :param expected_types: Tuple of types that correspond to expected function parameter types.
```

Comment:
I think this function should infer `expected_types` and `return_type` from the `name` argument.
I don't like that expected function signatures are supposed to be passed from callers.

Reply:
Should they be extracted to some kind of external dictionary to keep expected argument types and return type?
The idea was to make this method as freestanding as possible, to  make it able to validate many different types of methods.

Reply:
I think it would be better if mappings from function types to their signatures were defined in a single dictionary instead of being split across a hundred lines of code of `validate_script`.

If you would like for this method to remain freestanding, you can implement this with two functions -- the original `validate_callable` and a second one wrapping the first and acting as a proxy to fill out `expected_types` and `return_type` according to the dictionary of types.